tmt/steps/provision/__init__.py

```diff
@@ -335,6 +335,51 @@ class GuestData(tmt.utils.SerializableContainer):
         unserialize=lambda serialized: GuestFacts.from_serialized(serialized)
         )
 
+    def show(
+            self,
+            *,
+            keys: Optional[List[str]] = None,
+            verbose: int = 0,
+            logger: tmt.log.Logger) -> None:
+        """
+        Display guest data in a nice way.
+
+        :param keys: if set, only these keys would be shown.
+        :param verbose: desired verbosity. Some fields may be omitted in low
+            verbosity modes.
+        :param logger: logger to use for logging.
+        """
+
+        # Show empty data only in verbose mode
+        if self.is_bare and not verbose:
+            return
+
+        keys = keys or list(set(self.keys()))
+
+        def _emit_key(key: str) -> None:
+            # TODO: teach GuestFacts to cooperate with show() methods, honor
+            # the verbosity at the same time.
+            if key == 'facts':
+                return
+
+            value = getattr(self, key)
+
+            if value is None:
+                return
+
+            # TODO: it seems tmt.utils.format() needs a key, and logger.info()
+            # does not accept already formatted string.
+            if isinstance(value, (list, tuple)):
+                printable_value = fmf.utils.listed(value)
+
+            else:
+                printable_value = str(value)
+
+            logger.info(tmt.utils.key_to_option(key), printable_value, color='green')
+
+        for key in keys:
+            _emit_key(key)
```

Comment:
If the `for` loop is basically just an `_emit_key()` iterator why not move `for` to line 359?

Reply:
Fixed in 50e02b73